src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -332,7 +339,52 @@ async def create_docker_build_context(
     else:
         logger.debug("Did not build any packages for Docker image")
 
-    embedded_pkgs_digest = [built_package.digest for built_package in embedded_pkgs]
+    # Create digests for embedded packages. For upstream Docker images, we use only the image ID
+    # to ensure hash stability. This prevents changes in image tags (which may include timestamps
+    # or other dynamic values) from affecting the build context hash of dependent images.
+    embedded_pkgs_digest = []
+
+    # For Docker images, we need to extract the metadata filename and create stable digests
+    docker_metadata_gets = []
+    docker_package_indices = []
+    for i, built_package in enumerate(embedded_pkgs):
+        field_set = pkgs_wanting_embedding[i]
+        if request.build_upstream_images and isinstance(
+            getattr(field_set, "source", None), DockerImageSourceField
+        ):
+            docker_metadata_gets.append(Get(DigestContents, Digest, built_package.digest))
+            docker_package_indices.append(i)
+        else:
+            # For non-Docker packages, use the regular digest
+            embedded_pkgs_digest.append(built_package.digest)
+
+    # Get metadata contents for Docker images
+    if docker_metadata_gets:
+        docker_metadata_contents = await MultiGet(*docker_metadata_gets)
+
+        for metadata_contents, pkg_index in zip(docker_metadata_contents, docker_package_indices):
+            built_package = embedded_pkgs[pkg_index]
+
+            # Extract the original filename from the metadata
+            if metadata_contents:
+                original_filename = next(iter(metadata_contents)).path
```

Comment:
This is the name of the metadata file (some path ending in `docker-info.json`). I recommend munging the name, so it's clear when debugging that this is redacted data. E.g., `docker-info.stable.json`.